src/Catalyst.Core.Modules.Ledger/Ledger.cs

```diff
@@ -136,30 +154,80 @@ public void Update(Cid deltaHash)
 
         private void UpdateLedgerFromDelta(Cid deltaHash)
         {
-            if (!_synchroniser.DeltaCache.TryGetOrAddConfirmedDelta(deltaHash, out var nextDeltaInChain))
+            int stateSnapshot = _stateDb.TakeSnapshot();
+            int codeSnapshot = _codeDb.TakeSnapshot();
+            if (stateSnapshot != -1 || codeSnapshot != -1)
             {
-                _logger.Warning(
-                    "Failed to retrieve Delta with hash {hash} from the Dfs, ledger has not been updated.", deltaHash);
-                return;
+                if (_logger.IsEnabled(LogEventLevel.Error))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Merge this if statement with the enclosing one.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4350015)